openedx/core/djangoapps/theming/core.py

```diff
@@ -27,27 +32,20 @@ def comprehensive_theme_changes(theme_dir):
         'settings': {},
         'template_paths': [],
     }
-    root = Path(settings.PROJECT_ROOT)
-    if root.name == "":
-        root = root.parent
-
-    component_dir = theme_dir / root.name
 
-    templates_dir = component_dir / "templates"
-    if templates_dir.isdir():
-        changes['template_paths'].append(templates_dir)
+    root_name = get_project_root_name()
 
-    staticfiles_dir = component_dir / "static"
-    if staticfiles_dir.isdir():
-        changes['settings']['STATICFILES_DIRS'] = [staticfiles_dir] + settings.STATICFILES_DIRS
+    if themes_dir.isdir():
+        changes['template_paths'].append(themes_dir)
 
-    locale_dir = component_dir / "conf" / "locale"
-    if locale_dir.isdir():
-        changes['settings']['LOCALE_PATHS'] = [locale_dir] + settings.LOCALE_PATHS
+    for theme_dir in os.listdir(themes_dir):
+        staticfiles_dir = os.path.join(themes_dir, theme_dir, root_name, "static")
+        if staticfiles_dir.isdir():
+            changes['settings']['STATICFILES_DIRS'] = settings.STATICFILES_DIRS + [staticfiles_dir]
```

Comment:
Each theme is added to `STATICFILES_DIRS` since collectstatic would not run in request/response cycle and we would not know current site.